src/qml/bitcoin.cpp

```diff
@@ -144,11 +154,19 @@ int QmlGuiMain(int argc, char* argv[])
     }
 
     /// Read and parse settings.json file.
-    if (!gArgs.InitSettings(error)) {
+    std::vector<std::string> errors;
+    if (!gArgs.ReadSettingsFile(&errors)) {
+        error = strprintf("Failed loading settings file:\n%s\n", MakeUnorderedList(errors));
         InitError(Untranslated(error));
         return EXIT_FAILURE;
     }
 
+    QVariant need_onboarding(true);
+    if (gArgs.IsArgSet("-datadir")) {
```

Comment:
the datadir could be empty, in which case we'd want to still show onboarding to allow the user to configure